cadc-vosi/src/main/java/ca/nrc/cadc/vosi/AvailabilityServlet.java

```diff
@@ -55,22 +124,28 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
             log.info(logInfo.start());
 
             Class wsClass = Class.forName(pluginClassName);
-            WebService ws = (WebService) wsClass.newInstance();
+            AvailabilityPlugin ws = (AvailabilityPlugin) wsClass.newInstance();
             if (ws instanceof AvailabilityPlugin) {
                 AvailabilityPlugin ap = (AvailabilityPlugin) ws;
                 ap.setAppName(appName);
             }
             
-            AvailabilityStatus status = ws.getStatus();
-
-            Availability availability = new Availability(status);
-            availability.setClientIP(NetUtil.getClientIP(request));
-
-            Document document = availability.toXmlDocument();
-            XMLOutputter xop = new XMLOutputter(Format.getPrettyFormat());
-            started = true;
-            response.setContentType("text/xml");
-            xop.output(document, response.getOutputStream());
+            String detail = request.getParameter("detail");
+            if (detail != null && detail.equals("min")) {
```

Comment:
The normal getStatus() method returns the status and any thrown exception is a bug in getStatus(). However, the heartbeat() function is implicitly using a thrown RuntimeException to indicate "down" which is not a bug. We *do not* want every failed heartbeat() test to print a stack trace with "BUG: " in it.

So, existing design says that it AvailabilityPlugin method should be "boolean heartbeat()" and then any exception thrown is a bug.

Reply:
Also, I thought the plan was to delete the old (deprecated) interface so that services could not be rebuilt without changing to AvailabilityPlugin. Is there a flaw in that plan?

Reply:
Updated to:
public boolean heartbeat();

Removed WebService.java file.